### PR TITLE
Display the last-committed time as modified column when possible.

### PR DESCRIPTION
**Pull request by pelmers**
Fixes bug 1189576

I'm not sure if the tests depend on the location where it's running, it may be better to generate the tests from timestamps instead in that case, i.e. I record the expected timestamp in the test, then the test case constructs the locale-specific format for it at run time.

### CODE REVIEW

**Comment by pelmers** [comment]
Consider using <time> html tag.

File: dxr/vcs.py [hunk 1]
```diff
@@ -75,6 +76,12 @@ def generate_raw(self, path):
         """Construct URL to upstream view to raw file at path."""
         return NotImplemented
 
+    def last_modified_date(self, path):
+        """Return a datetime object that represents the last time a commit was
+        made to the given path.
+        """
+        return NotImplemented
```
Comment (pelmers) [inline dxr/vcs.py]: `raise notimplementederror`

**Comment by erikrose** [comment]
r+ with notimplementederror as exception and fix of testcase to work regardless of timezone

**Comment by pelmers** [comment]
Just realized that this change touches the shape of the file index, since 'modified' is a one-element list rather than a string.
This isn't worth a format bump, right? I can let it accept either the string or the list.

**Comment by erikrose** [comment]
Yes, just let it accept either. But comment that spot, and say that we can remove support for the non-list case when we hit format version X.

File: dxr/templates/folder.html [hunk 2]
```diff
@@ -35,7 +35,7 @@
       {% for icon, name, modified, size, relative_url in files_and_folders %}
         <tr>
           <td><a href="{{ relative_url }}" class="icon {{ icon }}">{{ name }}</a></td>
-          <td><a href="{{ relative_url }}">{{ '' if modified is none else modified.strftime("%Y %b %d %H:%m") }}</a></td>
+          <td><a href="{{ relative_url }}"><time>{{ '' if modified is none else modified.strftime("%Y %b %d %H:%m") }}</time></a></td>
```
Comment (erikrose) [inline dxr/templates/folder.html]: Probably we shouldn't have an empty time tag here in any case.

**Comment by erikrose** [comment]
r+ with empty time tags gone

File: dxr/templates/folder.html [hunk 3]
```diff
@@ -35,7 +35,11 @@
       {% for icon, name, modified, size, relative_url in files_and_folders %}
         <tr>
           <td><a href="{{ relative_url }}" class="icon {{ icon }}">{{ name }}</a></td>
-          <td><a href="{{ relative_url }}"><time>{{ '' if modified is none else modified.strftime("%Y %b %d %H:%m") }}</time></a></td>
+          <td><a href="{{ relative_url }}">
+            {% if modified is not none %}
+              <time>{{ '' if modified is none else modified.strftime("%Y %b %d %H:%m") }}</time>
```
Comment (erikrose) [inline dxr/templates/folder.html]: `modified` is not none, so it can't be none. :-)

**Comment by erikrose** [comment]
r+ if tests pass!

**Pull request merged**